File: 2020/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0, ans = Integer.MIN_VALUE;
        for (int n : nums) {
            sum += n;
            ans = Math.max(sum, ans);
            sum = Math.max(0, sum);
        }
        return ans;
    }
}

File: 2024/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, maxSum = INT_MIN;
        for (int n : nums) {
            sum += n;
            maxSum = max(maxSum, sum);
            sum = max(sum, 0);
        }
        return maxSum;
    }
};